src/mo_interpreter/interpret.ml

```diff
@@ -431,10 +431,15 @@ and interpret_exp_mut env exp (k : V.value V.cont) =
             match f.T.typ with
             | T.Func (s, c, b, a, r) ->
               let func = fun _ _ k ->
-                match (url_text, f.T.lab) with
-                | ("aaaaa-aa", "raw_rand") ->
-                  k (V.Blob (String.init 32 (fun _ -> Char.chr (Random.int 256))))
-                | _ ->  trap exp.at "unsupported method %s in actor \"%s\"" f.T.lab (V.as_text v1)
+                match (bytes, f.T.lab) with
+                | ("", "raw_rand") ->
```

Comment:
Is it possible for the IC canister to use a different URL from `aaaaa-aa`? Otherwise, we could maybe do something like this:

```suggestion
                match (bytes, url_text, f.T.lab) with
                | ("", "aaaaa-aa", "raw_rand") ->
```

Reply:
I believe "aaaaa-aa" is hardcoded into the interface spec of the IC.

Reply:
https://internetcomputer.org/docs/current/references/ic-interface-spec#ic-management-canister

Reply:
Okay, good to know. Although it's unlikely for anyone to run into this, it seems reasonable to restrict the `raw_rand` behavior to the management canister's principal (in case someone tries to use an unrelated `raw_rand` implementation from a different canister).